### ISSUE TITLE
[v2]  iOS's fallback gesture causes Navigation to be unavailable

### ISSUE THREAD

**Issue by iqysf** [wix/react-native-navigation#3576]
### Issue Description

This problem occurs when you use the back gesture after pop from the previous page.

[](url)

### Steps to Reproduce / Code Snippets / Screenshots

[video record](https://yiliandai-img.oss-cn-shenzhen.aliyuncs.com/QQ20180718-203029.mp4)

---
### Environment
* React Native Navigation version:2.0.2418
* React Native version: 0.56.0
* Platform(s): iOS
* Device info: Simulator && Device / iOS12 Developer Beta4&& iOS 11.4  / Debug&&Release

**Comment by ujwal-setlur** [comment]
Hmm, seems to work for me ok...can you show your code?

**Comment by iqysf** [comment]
@ujwal-setlur 
This example has the same problem：https://github.com/iqysf/rnndemo
video: [video](https://yiliandai-img.oss-cn-shenzhen.aliyuncs.com/QQ20180719-132855-HD.mp4)

The RNNRootViewController after the push was found in the View UI Hierarchy, but it was not displayed on the device.
![](https://yiliandai-img.oss-cn-shenzhen.aliyuncs.com/WX20180719-133122%402x.png)

**Comment by ujwal-setlur** [comment]
@iqysf i am a little confused. Where are you registering your screens? I guess you have only one screen that you push again, bit don't you need to register it?

**Comment by ujwal-setlur** [comment]
Oh I see it in index.js

**Comment by ujwal-setlur** [comment]
Yeah, I see the problem now. I can reproduce it in my app as well. It gets triggered when 

- first a root screen is displayed
- then, a second screen is pushed
- then, the second screen is popped back to the first screen
- then, back gesture is is done on first screen
- then, pushing second screen does not work anymore

**Comment by ujwal-setlur** [comment]
One thing I noticed is that if I do another back gesture on the first screen, a screen gets popped which really shouldn't have, but then the pushing works again...

**Comment by ujwal-setlur** [comment]
So, if you disable animations in the push, it resolves the problem:

```js
                    Navigation.push(this.props.componentId, {
                        component: {
                            name: 'App',
                            options: {
                                animations: {
                                    push: {
                                        enable: false
                                    }
                                }
                            }
                        }
                    });
```

So, this has something to do with animations...

**Comment by ujwal-setlur** [comment]
I believe this patch will fix the problem:

```diff
--- RNNNavigationStackManager.m	2018-07-19 00:46:01.000000000 -0700
+++ node_modules/react-native-navigation/lib/ios/RNNNavigationStackManager.m	2018-07-19 00:55:42.000000000 -0700
@@ -13,6 +13,7 @@
 	} else {
 		nvc.delegate = nil;
 		nvc.interactivePopGestureRecognizer.delegate = nil;
+		nvc.interactivePopGestureRecognizer.enabled = NO;
 	}

 	[self performAnimationBlock:^{
```

I tested this on both your app and mine.

**Comment by iqysf** [comment]
 @ujwal-setlur Yeah,I tried your patch and found no problems. thx！

**Comment by ujwal-setlur** [comment]
This is still an issue until it gets fixed in the official code. I will issue a pull request

**Comment by ujwal-setlur** [comment]
@iqysf, can you please reopen?

**Comment by iqysf** [comment]
@ujwal-setlur OK

**Comment by matpaul** [comment]
I think it's very high bug(

At this moment fix for me was comment it in RNNNavigationStackManager
nvc.interactivePopGestureRecognizer.delegate = nil;
and do not use any style/customization for leftButtons

**Comment by matpaul** [comment]
@yogevbd Hi can you check it? i pretty sure i can see same bug on v1

**Comment by Seven88** [comment]
same issue here

**Comment by Nualiian** [comment]
this is still not fixed for me. Seems to me we have two options as of now:
One is to use a version where the pop gesture is working with a bug, e.g. `2.1.3`.
The second option is to use a version without the pop gesture and hence without the bug, e.g. `2.2.3`.
Both of these options are not ideal.
